app/src/editor/shared/components/modules/Scheduler.jsx

```diff
@@ -0,0 +1,714 @@
+    },
+}
+
+const addRule = (rules, id = uuid()) => ([
+    ...rules,
+    {
+        id,
+        ...defaultRule,
+    },
+])
+
+const removeRule = (rules, id) => rules.filter((rule) => rule.id !== id)
+
+const changeRule = (rules, id, properties) => rules.map((r) => {
+    if (r.id === id) {
+        return {
+            ...r,
+            ...properties,
+        }
+    }
+
+    return r
+})
+
+export default class SchedulerModule extends React.Component {
+    subscription = React.createRef()
+
+    state = {
+        defaultValue: undefined,
+        rules: undefined,
+    }
+
+    static getDerivedStateFromProps(props, state) {
+        const checksum = JSON.stringify(props.module.schedule.rules)
+        if (state.rules && state.checksum === checksum) {
+            return null
+        }
+
+        const { rules, defaultValue } = props.module.schedule || {}
+
+        // Add default row when module is added to canvas
+        const existingRules = rules || []
+        const defaultRules = (!existingRules.length && !state.rules) ? [{
+            ...defaultRule,
+        }] : existingRules
+
+        // Map existing rules with an id
+        return {
+            defaultValue: defaultValue || 0,
+            rules: defaultRules.map((rule) => ({
+                ...rule,
+                id: rule.id || uuid(),
+            })),
+            checksum,
+        }
+    }
+
+    onAddRule = () => {
+        this.setState((prevState) => ({
+            rules: addRule(prevState.rules),
+        }), this.updateStateToModule)
+    }
+
+    onRemoveRule = (id) => {
+        this.setState((prevState) => ({
+            rules: removeRule(prevState.rules, id),
+        }), this.updateStateToModule)
+    }
+
+    onChangeRule = (id, properties) => {
+        this.setState((prevState) => ({
+            rules: changeRule(prevState.rules, id, properties),
+        }), this.updateStateToModule)
+    }
+
+    onDefaultValueChange = (defaultValue) => {
+        this.setState({
+            defaultValue,
+        }, this.updateStateToModule)
+    }
+
+    updateStateToModule = () => {
+        this.props.api.updateModule(this.props.moduleHash, {
+            schedule: {
+                ...this.state,
+            },
+        })
+    }
+
+    onSortEnd = ({ newIndex, oldIndex }) => {
+        this.setState(({ rules }) => ({
+            rules: arrayMove(rules, oldIndex, newIndex),
+        }), this.updateStateToModule)
+    }
+
+    render() {
+        const { module, isActive } = this.props
+        const { rules, defaultValue } = this.state
+
+        return (
+            <div>
+                <ModuleSubscription
+                    {...this.props}
+                    ref={this.subscription}
+                    module={module}
+                />
+                <SortableList
+                    distance={1} /* This will allow clicks to pass through */
+                    onSortEnd={this.onSortEnd}
+                    lockAxis="y"
```

Comment:
Wondering if the sortable items should use a hand cursor, currently the sortability of the list isn't very discoverable IMO.